collect_app/src/test/java/org/odk/collect/android/widgets/base/GeneralExStringWidgetTest.java

```diff
@@ -64,6 +66,26 @@ public void widgetShouldBeRegisteredForContextMenu() {
         assertThat(viewsRegisterForContextMenu.get(1).getId(), is(widget.getId()));
     }
 
+    @Test
+    public void errorDisappearsOnSetData() {
```

Comment:
We should have such tests for other question types. Here is a separate issue for that https://github.com/getodk/collect/issues/6274.